modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/endpoints/AbstractResourceManager.java

```diff
@@ -67,9 +67,13 @@ public static JSONDecoder getDecoder() throws CharonException {
      * @return endpoint URL
      * @throws NotFoundException
      */
-    public static String getResourceEndpointURL(String resource) throws NotFoundException {
+    public static String getResourceEndpointURL(String resource, String context) throws NotFoundException {
         if (endpointURLMap != null && endpointURLMap.size() != 0) {
-            return endpointURLMap.get(resource);
+            String endpointURL = endpointURLMap.get(resource);
+            if (context == null)
```

Comment:
Rather than just validating for a null value better to validate if the param 'context' is an empty value.